binary/private/binary_roche_deformation.f90

```diff
@@ -0,0 +1,357 @@
+! ***********************************************************************
+!
+!   Copyright (C) 2025 Matthias Fabry & The MESA Team
+!
+!   MESA is free software; you can use it and/or modify
+!   it under the combined terms and restrictions of the MESA MANIFESTO
+!   and the GNU General Library Public License as published
+!   by the Free Software Foundation; either version 2 of the License,
+!   or (at your option) any later version.
+!
+!   You should have received a copy of the MESA MANIFESTO along with
+!   this software; if not, it is available at the mesa website:
+!   http://mesa.sourceforge.net/
+!
+!   MESA is distributed in the hope that it will be useful,
+!   but WITHOUT ANY WARRANTY; without even the implied warranty of
+!   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+!   See the GNU Library General Public License for more details.
+!
+!   You should have received a copy of the GNU Library General Public License
+!   along with this software; if not, write to the Free Software
+!   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+!
+! ***********************************************************************
+
+module binary_roche_deformation
+
+   ! computes the stellar structure correction factors fp/ft and the specific
+   ! moment of inertia i_rot assuming shellularity in the Roche potential of
+   ! a binary star. Follows Fabry, Marchant and Sana, 2022, A&A 661, A123.
+
+   use ieee_arithmetic, only: ieee_is_nan
+   use interp_2d_lib_db
+   use auto_diff
+   use star_def
+   use binary_def
+   use binary_utils, only : eval_rlobe
+
+
+   implicit none
+
+   real(dp), parameter :: nudge = 1d-4
+   real(dp), pointer :: xvals(:), yvals(:), yvals_gtr_than_1(:), fpfunc1d(:), ftfunc1d(:), &
+         irotfunc1d(:), otherrfunc1d(:), afunc1d(:)
+   logical :: inter_ok = .false., dbg = .true.
```

Comment:
I'm guessing we don't generally want the debug output on, otherwise it isn't really debug output.
```suggestion
   logical :: inter_ok = .false., dbg = .false.
```